tls_utils.go

```diff
@@ -4,36 +4,65 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"io/ioutil"
+	"runtime"
 
 	"github.com/pkg/errors"
 )
 
 // GetClientTLSConfig creates a creates a client-side TLS configuration based
-// on the given ca, cert, and key.
-func GetClientTLSConfig(ca, crt, key []byte) (*tls.Config, error) {
-	cp := x509.NewCertPool()
-	if !cp.AppendCertsFromPEM(ca) {
-		return nil, errors.New("credentials: failed to append certificates")
+// on the given cas, cert, and key. If possible, the system cert pool is
+// combined with the provided cas.
+func GetClientTLSConfig(cas [][]byte, crt, key []byte) (*tls.Config, error) {
+	cp, err := GetCACertPool(cas...)
+	if err != nil {
+		return nil, err
 	}
 
 	keyPair, err := tls.X509KeyPair(crt, key)
 	if err != nil {
-		return nil, errors.Wrap(err, "problem reading client cert")
+		return nil, errors.Wrap(err, "reading the client cert")
 	}
 
 	return &tls.Config{
 		Certificates: []tls.Certificate{keyPair},
 		RootCAs:      cp,
 	}, nil
+}
+
+// GetCACertPool is a convenience function that creates a cert pool with the
+// given cas and, when possible, the system cert pool. On windows machines, the
+// cert pool will be empty if no cas are passed in.
+func GetCACertPool(cas ...[]byte) (*x509.CertPool, error) {
+	cp := x509.NewCertPool()
+	if runtime.GOOS != "windows" {
```

Comment:
nit: rather than making an if switch on `GOOS`, can we just make a `tls_utils_windows.go` file with a separate `GetCACertPool` implementation that doesn't include the `SystemCertPool` logic?

Reply:
I would almost rather leave it here so the documentation is clearer. I also hope that this will eventually get fixed in future versions of go and we can remove it. But if you are opinionated about it I am fine moving it.

Reply:
Eh, I don't think it's that important, so LGTM.